### PR TITLE
 home-assistant-custom-components.waste_collection_schedule: init at 1.44.0

### PR DESCRIPTION
**Pull request by JamieMagee**
## Description of changes

Adds the `icalevents` Python package and `waste_collection_schedule` Home Assistant custom component.

There is one patch applied to `waste_collection_schedule` that is in PR upstream: https://github.com/mampfes/hacs_waste_collection_schedule/pull/1515

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JamieMagee** [comment]
Result of `nixpkgs-review pr 273717` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.icalevents</li>
    <li>python310Packages.icalevents.dist</li>
    <li>python311Packages.icalevents</li>
    <li>python311Packages.icalevents.dist</li>
  </ul>
</details>

**Comment by dotlambda** [comment]
> This is a dependency of the `home-assistant-custom-component.waste_collection_schedule` package that I currently have [in a branch](https://github.com/JamieMagee/nixpkgs/commit/7d94687da1fb3fda550cb92c80a04c659f1dbbfc). Though that package is also blocked on an issues upstream: [mampfes/hacs_waste_collection_schedule#1515](https://github.com/mampfes/hacs_waste_collection_schedule/pull/1515)

I think you should include that package in this PR. That makes it easier to test. You can use `fetchpatch` to apply the required changes.

File: pkgs/development/python-modules/icalevents/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    pytestCheckHook
+  ];
+
+  disabledTests = [
+    # Makes HTTP calls
+    "test_events_url"
+    "test_events_async_url"
+  ];
+
+  pythonImportsCheck = ["icalevents"];
```
Comment (dotlambda) [inline pkgs/development/python-modules/icalevents/default.nix:50]:
```suggestion
  pythonImportsCheck = [ "icalevents" ];
```

File: pkgs/development/python-modules/icalevents/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  meta = with lib; {
+    changelog = "https://github.com/jazzband/icalevents/releases/tag/v${version}";
+    description = "Python module for iCal URL/file parsing and querying";
+    homepage = "https://github.com/jazzband/icalevents";
+    maintainers = with maintainers; [jamiemagee];
```
Comment (dotlambda) [inline pkgs/development/python-modules/icalevents/default.nix:56]:
```suggestion
    maintainers = with maintainers; [ jamiemagee ];
```

File: pkgs/development/python-modules/icalevents/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  fetchFromGitHub,
+  pythonOlder,
+  pytestCheckHook,
+  poetry-core,
+  datetime,
+  httplib2,
+  icalendar,
+  python-dateutil,
+  pytz,
+}:
```
Comment (dotlambda) [inline pkgs/development/python-modules/icalevents/default.nix:13]:
```suggestion
{ lib
, buildPythonPackage
, fetchFromGitHub
, pythonOlder
, pytestCheckHook
, poetry-core
, datetime
, httplib2
, icalendar
, python-dateutil
, pytz
}:

```

**Comment by JamieMagee** [comment]
> I think you should include that package in this PR. That makes it easier to test. You can use `fetchpatch` to apply the required changes.

Thanks for the suggestion. I hoped the upstream PR would be merged quickly, but nixpkgs maintainers are quicker 😉 

Regarding the style changes, is that to bring it in line with RFC166?

**Comment by JamieMagee** [comment]
Result of `nixpkgs-review pr 273717` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-custom-components.waste_collection_schedule</li>
    <li>python310Packages.icalevents</li>
    <li>python310Packages.icalevents.dist</li>
    <li>python311Packages.icalevents</li>
    <li>python311Packages.icalevents.dist</li>
  </ul>
</details>

File: pkgs/servers/home-assistant/custom-components/default.nix [hunk 4]
```diff
@@ -6,6 +6,8 @@
 
   govee-lan = callPackage ./govee-lan {};
 
+  waste_collection_schedule = callPackage ./waste_collection_schedule {};
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/default.nix:9]: Keep this attrset sorted.

**Comment by JamieMagee** [comment]
Thanks for the explanation. I've made the changes locally and force pushed to keep the commits separate.

**Comment by JamieMagee** [comment]
Result of `nixpkgs-review pr 273717` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-custom-components.waste_collection_schedule</li>
    <li>python310Packages.icalevents</li>
    <li>python310Packages.icalevents.dist</li>
    <li>python311Packages.icalevents</li>
    <li>python311Packages.icalevents.dist</li>
  </ul>
</details>

**Review by mweinelt — APPROVED** [review]

File: pkgs/development/python-modules/icalevents/default.nix [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+, pythonOlder
+, pytestCheckHook
+, poetry-core
+, datetime
+, httplib2
+, icalendar
+, python-dateutil
+, pytz
+}:
+buildPythonPackage rec {
```
Comment (dotlambda) [inline pkgs/development/python-modules/icalevents/default.nix:13]:
```suggestion

buildPythonPackage rec {
```

File: pkgs/servers/home-assistant/custom-components/waste_collection_schedule/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  buildHomeAssistantComponent,
```
Comment (dotlambda) [inline pkgs/servers/home-assistant/custom-components/waste_collection_schedule/default.nix:3]:
```suggestion
{ lib,
, buildHomeAssistantComponent
```

**Review by dotlambda — APPROVED** [review]
I don't like the style but if @mweinelt doesn't mind then it's fine with me.

**Comment by JamieMagee** [comment]
@dotlambda I made the formatting changes you requested. IMO this is ready for merge.

**Comment by JamieMagee** [comment]
Result of `nixpkgs-review pr 273717` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-custom-components.waste_collection_schedule</li>
    <li>python310Packages.icalevents</li>
    <li>python310Packages.icalevents.dist</li>
    <li>python311Packages.icalevents</li>
    <li>python311Packages.icalevents.dist</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
> I don't like the style but if @mweinelt doesn't mind then it's fine with me.

I mostly lack the energy to repeatedly comment on stylistic issues. When I touch the package I usually fix it, because it is less of a hassle for me personally. Really hoping for the formatter RFC to conclude soon.

**Pull request merged**